packages/taquito-michelson-encoder/src/michelson-map.ts

```diff
@@ -19,21 +21,52 @@ const michelsonMapTypeSymbol = Symbol.for('taquito-michelson-map-type-symbol');
 
 export type MichelsonMapKey = Array<any> | object | string | boolean | number;
 
-const isMapType = (
-  value: MichelsonV1Expression
-): value is { prim: 'map' | 'big_map'; args: [MichelsonV1Expression, MichelsonV1Expression] } => {
-  return 'args' in value && Array.isArray(value.args) && value.args.length === 2;
-};
+/**
+ *
+ * @throws {@link InvalidMapTypeError} when the argument passed to mapType is not a valid map type
+ */
+function validateMapType(value: MichelsonV1Expression): asserts value is {
+  prim: 'map' | 'big_map';
+  args: [MichelsonV1Expression, MichelsonV1Expression];
+} {
+  if (!('prim' in value)) {
+    throw new InvalidMapTypeError(value, `Missing 'prim' field`);
+  }
+  if (!['map', 'big_map'].includes(value.prim)) {
+    throw new InvalidMapTypeError(value, `The prim field should be 'map' or 'big_map'`);
+  }
+  if (!('args' in value)) {
+    throw new InvalidMapTypeError(value, `Missing 'args' field`);
+  }
+  if (!Array.isArray(value.args)) {
+    throw new InvalidMapTypeError(value, `The 'args' field should be an array`);
+  }
+  if (value.args.length !== 2) {
+    throw new InvalidMapTypeError(value, `The 'args' field should have 2 elements`);
+  }
+}
 
 /**
  *  @category Error
  *  @description Error that indicates a map type mismatch, where an attempt to set a key or value in a Map doesn't match the defined type of the Map
  */
-export class MapTypecheckError extends Error {
+export class MapTypecheckError extends TaquitoError {
   name = 'MapTypecheckError';
 
-  constructor(public readonly value: any, public readonly type: any, errorType: 'key' | 'value') {
-    super(`${errorType} not compliant with underlying michelson type`);
+  constructor(
+    public readonly value: any,
+    public readonly type: any,
+    objectType: 'key' | 'value',
+    public readonly reason: any
+  ) {
+    super(
```

Comment:
might still have some super(message) in places, hope you don't mind updating them